File: 3005-count-elements-max-frequency/solution.java
public int maxFrequencyElements(int[] nums) {
        return Arrays.stream(nums).mapToObj(x -> Arrays.stream(nums).filter(i -> i == x)).mapToInt(j -> (int)j.count()).map(k -> k == Arrays.stream(nums).mapToObj(n -> Arrays.stream(nums).filter(y -> y == n)).mapToInt(z -> (int)z.count()).max().orElse(0) ? 1 : 0).sum();
}

File: 3005-count-elements-max-frequency/solution.c
int maxFrequencyElements(int* nums, int numsSize) {
    int freq[102] = {0};
    for (int i = 0; i < numsSize; i++) freq[100] += ++freq[nums[i]-1] > freq[100];
    for (int i = 0; i < 100; i++) freq[101] += freq[100] * (freq[i] == freq[100]);
    return freq[101];
}
